app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+
+  end
+
+  def create
+    user = User.find_by(username: params[:session][:username])
+    if user && user.authenticate(params[:session][:password])
+      session[:user_id] = user.id
+      redirect_to dashboard_path
+    else
+      flash[:error] = "Invalid username or password"
+      redirect_to login_path
+    end
+  end
+
+  def destroy
```

Comment:
When a user logs out, should we also delete their shopping cart session? If another person uses the same computer and logs in, they will automatically have everything from the first user's cart populate their cart.